integration-tests/dataformats-json/src/main/java/org/apache/camel/quarkus/component/dataformats/jackson/json/JacksonJsonResource.java

```diff
@@ -0,0 +1,692 @@
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.fasterxml.jackson.module.jaxb.JaxbAnnotationModule;
+import org.apache.camel.CamelContext;
+import org.apache.camel.ConsumerTemplate;
+import org.apache.camel.Exchange;
+import org.apache.camel.ExchangePattern;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.component.jackson.JacksonConstants;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.quarkus.component.dataformats.json.model.DummyObject;
+import org.apache.camel.quarkus.component.dataformats.json.model.Order;
+import org.apache.camel.quarkus.component.dataformats.json.model.Pojo;
+import org.apache.camel.quarkus.component.dataformats.json.model.TestJAXBPojo;
+import org.apache.camel.quarkus.component.dataformats.json.model.TestOtherPojo;
+import org.apache.camel.quarkus.component.dataformats.json.model.TestPojo;
+import org.apache.camel.quarkus.component.dataformats.json.model.TestPojoView;
+import org.apache.camel.support.DefaultExchange;
+import org.jboss.logging.Logger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Path("/dataformats-json")
+@ApplicationScoped
+public class JacksonJsonResource {
+
+    private static final Logger LOG = Logger.getLogger(JacksonJsonResource.class);
+    private static final String LS = System.lineSeparator();
+    @Inject
+    ProducerTemplate producerTemplate;
+    @Inject
+    ConsumerTemplate consumerTemplate;
+    @Inject
+    CamelContext context;
+
+    @Path("jackson/unmarshal-typeheader")
+    @GET
+    public void jacksonXmlUnmarshalTypeHeader(String body) throws Exception {
+
+        MockEndpoint mock = context.getEndpoint("mock:jackson-unmarshal-type-header-reversePojo", MockEndpoint.class);
+        mock.expectedMessageCount(1);
+        mock.message(0).body().isInstanceOf(TestPojo.class);
+
+        String json = "{\"name\":\"Camel\"}";
+        producerTemplate.sendBodyAndHeader("direct:jackson-unmarshal-type-header-backPojo", json,
+                JacksonConstants.UNMARSHAL_TYPE, TestPojo.class.getName());
+
+        mock.assertIsSatisfied();
+
+        TestPojo pojo = mock.getReceivedExchanges().get(0).getIn().getBody(TestPojo.class);
+        assertNotNull(pojo);
+        assertEquals("Camel", pojo.getName());
+    }
+
+    @Path("jackson/unmarshal-list")
+    @GET
+    public void jacksonUnmarshalList(String body) throws Exception {
+
+        MockEndpoint mock = context.getEndpoint("mock:jackson-unmarshal-reversePojo", MockEndpoint.class);
```

Comment:
What about renaming the mock to `mock:jackson-unmarshal-backPojo` so that one could find the match direct route easily ?

Reply:
Yes, I will revise this accordingly.